Bootstrap

manny opened the pull request:
Still getting the hang of this pull request stuff. Accidentally sent one on the master branch. Fixed the things you had brought up

Commit:
login for blog post

Commit:
added bootstrap files. found registration bug in register user method

Commit:
fixed register bug

Commit:
added test file to bootstrap branch

Commit:
barebones login functionality complete. on to bootstraping

Commit:
testing branching

Commit:
commiting to new branch

Commit:
pushing to new branch

Commit:
ready for bootstraping

Commit:
got bootstrap working
fixed typos in tags

Commit:
auto increment user ID

Commit:
added viewpost, post, and style

    users can now make post, which enter into the database
    added links were I said I would
    threw in alittle style to make it less hideous

JarekSed commented on common.php:
```diff
@@ -0,0 +1,40 @@
+<?php
+require_once("database.php");
+session_start();
+function getUserName(){
```
You need some sort of comment on all functions. At least: what arguments does it take, what does it do, and what does it return.

JarekSed commented on common.php:
```diff
@@ -0,0 +1,40 @@
+    }
+}
+
+function getID(){
```
Function needs comments

JarekSed commented on common.php:
```diff
@@ -0,0 +1,40 @@
+    global $db;  
+    if(!isset($_SESSION['username'])){
+        return false;
+        //link to login page
```
This doesn't link to the login page, so remove the comment

JarekSed commented on index.php:
```diff
@@ -0,0 +1,43 @@
+
+<body>  
+    <?php
+    require_once("common.php");  // creates session array(kinda like POST) that can save  variables between pages
```
This comment is out of date, and wrong.

JarekSed commented on index.php:
```diff
@@ -0,0 +1,43 @@
+        <input type = "submit" name = "logoutButton" value = "Log Out"/>
+        </form>
+    <?php
+    //bootstrap branch
```
Useless comment

JarekSed commented on login.php:
```diff
@@ -0,0 +1,32 @@
+<?php
+    require("database.php");
```
require("database.php") and session_start() with requiring common.php. 
Let's put all initialization code and common functions in common.php, and just include it everywhere

JarekSed commented on login.php:
```diff
@@ -0,0 +1,32 @@
+<?php
+    require("database.php");
+    session_start();
+    if(!isset($_POST["username"])){
```
Wouldn't hurt to put a comment here, saying why we redirect when username is not set.

JarekSed commented on login.php:
```diff
@@ -0,0 +1,32 @@
+    if(!isset($_POST["username"])){
+        header("Location: http://jsedlacek.info/~mjl288/Manny-Blog/index.php");
+    }
+    if(isset($_POST["logoutButton"])){
```
comment why we do this

JarekSed commented on login.php:
```diff
@@ -0,0 +1,32 @@
+        unset($_SESSION['loginFail']);
+        unset($_POST['logoutButton']);
+        header("Location: http://jsedlacek.info/~mjl288/Manny-Blog/index.php");
+    }else{
```
comment when this else block gets use

JarekSed commented on register.php:
```diff
@@ -0,0 +1,71 @@
+<?php
+    require_once("common.php");
+?>
```
Should we let you view this page if you are already logged in? This is really a matter of opinion, to me it seems weird to register while logged in. We should prolly make a check at the top of this page, if you are already logged in just print a message saying "you are already logged into an account, log out to register a new one" w/ a link back to home or something.

JarekSed commented on test:
```diff
@@ -0,0 +1 @@
+testing branching
```
You can delete this test file.

JarekSed commented:
Code all looks good. You need comments in a few places, but the code seems solid. This is definitely a good start for the project.

Commit:
fixed problems address in pull request comments

Commit:
fixed issues addressed in pull request comments

    -commented functions and important control flows
    -added link getID function for redirect
    -removed out of date comments
    -put all initialization code in common.php
    -added check to register.php that restricts and redirects logged in users from registering
    -deleted test file

JarekSed commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+*.swp
+*.swo
```
What is the .swo for?

JarekSed commented on common.php:
```diff
@@ -0,0 +1,43 @@
+require_once("database.php");
+session_start();  // creates session array(kinda like POST) that can save  variables between pages
+
+function getUserName(){  //returns name of logged in user, false if not logged in
```
The conventional way to do it is to use multiline comments, and put it above the function definition. For example:
/*
- This function does Foo.
- It takes 1 argument: Bar, an integer meaning whatever
- It returns true on success, and false on failure
  */
  function Bar($Foo){ ...

Commit:
added multiline commnts to above definitions
    -also took out a print statement inside the register function
